graphics/nanogui/Portfile

```diff
@@ -0,0 +1,193 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           cmake 1.1
+
+github.setup        mitsuba-renderer nanogui 21e5cbc880b2e26b28b2a35085a9e6706da1e2a8
+version             20200404-[string range ${github.version} 0 7]
+categories          graphics
+platforms           darwin
+license             BSD
+maintainers         @jasonliu-- openmaintainer
+description         minimalistic C++/Python GUI library for OpenGL, \
+                    GLES2/3, Metal, and WebAssembly/WebGL
+long_description    NanoGUI is a minimalistic cross-platform widget \
+                    library for OpenGL 3+, GLES 2/3, and Metal. It \
+                    supports automatic layout generation, stateful C++ \
+                    lambdas callbacks, a variety of useful widget \
+                    types, and Retina-capable rendering on Apple \
+                    devices.
+
+checksums           rmd160  192baaa0a7f0ac862b2d09021daa98733e342932 \
+                    sha256  f105aa23476f99c2db1b28bccf4751498584e366a30e5e5078d6963a8bab6c93 \
+                    size    2911997
+
+depends_lib-append  port:glfw \
+                    port:nanovg
+
+compiler.cxx_standard 2017
+configure.cxxflags-append -std=c++17
+
+post-patch {
+    # By default, NanoGUI's code assumes that the source codes of its
```

Comment:
there's a lot going on in this post-path block. I'll assume its all necessary, but it would be good to look to minimise there where we can in the future. i.e. if upstream needs to improve their build system for macOS, please feed this back to them.

Reply:
Yes, most of these patches should really be sent upstream. They use _a lot_ of hardcoded paths in their CMake file.

Reply:
OK. As you are the maintainer its up to you to maintain these fixes, but please do look to pushing these upstream as it will make it easier to maintainer going forward if the complexity here can be reduced.

Reply:
Yes, definitely agree. I'm sort of annoyed by how "messy" this file looks compared to most of my other portfiles. I don't even think my Blender portfile has this many changes, and Blender's source code is many times the size of NanoGUI. Frankly, I don't think that it should be MacPorts' role to fix up others' code before it becomes usable.

Reply:
no worries. I left this for a while to see if anyone else wanted to comment, but lets just merge now. If you are happy to maintain it thats fine.